main.cpp

```diff
@@ -66,18 +66,22 @@ struct MbedClientDevice device = {
 MbedClient mbed_client(device);
 
 
-// In case of K64F board , there is button resource available
-// to change resource value and unregister
-#ifdef TARGET_K64F
+// If board has a button_1, use it to update the counter.
+#ifdef BUTTON1
```

Comment:
It seems that the BUTTON1 and BUTTON2 identifiers are actually enum values (see PinNames.h:208 for TARGET_K64F) , so the preprocessor has no knowledge of them. Maybe they should be behind a macro where we could collect other standard definitions like the buttons?